### ISSUE TITLE
Wrong canonical on homepage

### ISSUE THREAD

**Issue by vladoa** [ahmedkaludi/accelerated-mobile-pages#4925]
I have WP installed in a separate directory ( `/wordpress/` ) and this is the canonical that is being displayed on the homepage AMP url ( `domain.com/?amp` ):

`domain.com/wordpress/`

instead of:

`domain.com/`

Screenshot:
https://imgur.com/a/sM118YN

WP Version: WordPress 5.6
Plugin Version: 1.0.71

**Comment by Zabi09** [comment]
@vladoa I have checked your issue on my end but I am unable to recreate your issue as the main directory canonical is only appearing on my end on the AMP homepage.

Will you please install the latest version of our plugin check once.

**Comment by vladoa** [comment]
Okay @Zabi09 I will update WP and the Plugin to the latest version and let you know.
Just to make sure, you tried moving WP to a different directory, like so example.com/wordpress/ ?

**Comment by Zabi09** [comment]
@vladoa Yes have tried with that my main directory is http://localhost/wordpress and then have created separately with the name another and then also the canonical is appearing fine.

Main directory - https://monosnap.com/file/3fSmQxycTznD1lWUO4WuujEC4fEnVE
Separate directory - https://monosnap.com/file/3gOk3mSgTUZIiIHhVUn3k9nGGzlmJ5

**Comment by vladoa** [comment]
Well yes that seems wrong.
In your Wordpress Settings > General, the Wordpress Address (URL) should be: http://localhost/wordpress
but your web address should be: http://localhost

Screenshot: https://imgur.com/a/kDewCg1

with settings like this, your public address of the website is http://localhost
and you access the AMP version like http://localhost/?amp or http://localhost/amp/
not like http://localhost/wordpress and the AMP version shouldn't be http://localhost/wordpress/?amp but http://localhost/?amp

Your WordPress Address (URL) is the address where WordPress files and folders are stored including your admin pages, media files, plugins, themes, etc.

Your Site Address (URL) in WordPress is the public facing part of your website. This is what your visitors will type in to reach your website.

The "Wordpress Address (URL)" is telling WP where it is installed, it shouldn't have anything to do with the public URLs and canonicals.

**Comment by Zabi09** [comment]
@vladoa By default, the WordPress Address and Site Address will be the same for all the websites. See this screenshot - https://prnt.sc/zxqo9p

Will you please tell me your use case why you are using the different URL for the site address and also will be please help us to recreate the same on our end so that we can easily recreate and help you to fix it

**Comment by vladoa** [comment]
Hey @Zabi09 , yes they are the same by default. 
Here is how to give Wordpress its own directory: https://wordpress.org/support/article/giving-wordpress-its-own-directory/
I used the [method 2](https://wordpress.org/support/article/giving-wordpress-its-own-directory/#method-ii-with-url-change) to do it, as described on the link.

The reasons can vary - in my case the reason is that I have a deployment script and it installs Wordpress, the Themes and the Plugins as Composer dependencies, keeping the directory structure clean instead of having WP core files cluttering the directory.

So basically you should just use the Site URL, not the WordPress URL to generate the canonicals in your AMP Plugin. It should be fairly easy to fix?

**Comment by Zabi09** [comment]
@vladoa I have created a custom plugin for fixing your concern, Please add this code by creating a new plugin and it will fix your issue.

`add_filter('ampforwp_modify_rel_url','vladoa_canonical');
function vladoa_canonical($url){
	if (ampforwp_is_home() || ampforwp_is_front_page()) {
		$url = get_site_url();
		return $url;
	}
	return $url;
}`

Please add it and let us know

**Comment by Zabi09** [comment]
@vladoa Have you added the code? is it working or not? Please let us know

**Comment by vladoa** [comment]
Hey @Zabi09 yes it is working, only thing I had to change was: `ampforwp_modify_rel_url` to `ampforwp_modify_rel_canonical`.
